src/main/java/com/puppycrawl/tools/checkstyle/utils/CommonUtils.java

```diff
@@ -467,4 +467,31 @@ public static String getFileExtension(String fileNameWithExtension) {
         }
         return extension;
     }
+
+    /**
+     * Checks whether the given string is a valid identifier.
+     * @param str A string to check.
+     * @return true when the given string contains valid identifier.
+     */
+    public static boolean isIdentifier(String str) {
+        boolean isIdentifier = !str.isEmpty() && Character.isJavaIdentifierStart(str.charAt(0));
```

Comment:
@soon My bad. They looked so similar I didn't notice one was `Start` and the other was `Part`.
Your example at https://github.com/checkstyle/checkstyle/pull/3924#discussion_r107225709 is fine then.